packages/migrate/Migration.js

```diff
@@ -11,11 +10,9 @@ const ResolverIntercept = require("./ResolverIntercept");
 const { getTruffleDb } = require("@truffle/db-loader");
 
 class Migration {
-  constructor(file, reporter, config) {
+  constructor(file, config) {
```

Comment:
Do we need an `expect.options(config, ["events"])` here?

Reply:
Alternatively we could define an `emit` method inside `Migration` here, like I've suggested for `Deployment`... and then we could put the `if` guard there and have them both conform to each other. (I don't think we want to have Migration call `deployer.emit`)

Reply:
yeah in general I don't oppose the idea of having emit methods in each of these classes/utilities - I'll go ahead and implement them

Reply:
Cool, then the original comment here about needing `expect.options` should really only apply if we want to keep the missing `if` statement that I call out [here](https://github.com/trufflesuite/truffle/pull/4436#discussion_r779196154).